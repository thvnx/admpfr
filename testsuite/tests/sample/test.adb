with Ada.Text_IO; use Ada.Text_IO;
with Admpfr; use Admpfr;

procedure Test is
   S, T, U : Mpfloat (200);
begin
   T.Set (Long_Float (1.0), RNDD);
   S.Set (Long_Float (1.0), RNDD);

   for I in 1 .. 100 loop
      T.Mul (T, Long_Integer (I), RNDU);
      U.Set (Long_Float (1.0), RNDD);
      U.Div (U, T, RNDD);
      S.Add (S, U, RNDD);
   end loop;

   Put ("Sum is ");

   Out_Str (Standard_Output, S, 10, RNDD);
   Put (ASCII.LF);
end Test;
