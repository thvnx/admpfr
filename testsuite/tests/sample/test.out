Sum is 2.7182818284590452353602874713526624977572470936999595749669131e0

